frontend/model/notifications/templates.js

```diff
@@ -1,133 +1,99 @@
-    read: true
-  }
-}
+type NewProposalType =
+  | 'ADD_MEMBER'
+  | 'CHANGE_MINCOME'
+  | 'CHANGE_VOTING_RULE'
+  | 'REMOVE_MEMBER';
 
-export const MEMBER_LEFT = ({ timestamp, data }: TemplateArguments): TemplateDescription => {
-  const name = data.username
-  const avatar = ''
-  return {
-    avatar,
-    icon: 'icon-minus',
-    level: 'info',
-    body: L('{name} has left your group. Contributions were updated accordingly.', {
-      name: `<strong>${name}</strong>`
-    }),
-    linkTo: '/contributions',
-    date: timeSince(timestamp, dateNow),
-    read: true
-  }
-}
-
-export const INCOME_DETAILS_OLD = ({ timestamp }: TemplateArguments): TemplateDescription => {
-  const avatar = ''
-  return {
-    avatar,
-    icon: 'coins',
-    level: 'info',
-    body: L("You haven't updated your income details in more than 6 months. Would you like to review them now?"),
-    linkTo: '/contributions/TODO-LINK-MODAL',
-    date: timeSince(timestamp, dateNow),
-    read: true
-  }
-}
+type NotificationLevel = 'danger' | 'info';
 
-export const SEND_CONTRIBUTION = ({ timestamp, data }: TemplateArguments): TemplateDescription => {
-  const month = 'June'
-  const avatar = ''
-  return {
-    avatar,
-    icon: 'coins',
-    level: 'info',
-    // REVIEW @mmbotelho - ranem contribution to payments?
-    body: L('Do not forget to send your {month} contributions.', {
-      month: `<strong>${month}</strong>`
-    }),
-    linkTo: '/payments',
-    date: timeSince(timestamp, dateNow),
-    read: true
-  }
+export type NotificationTemplate = {
+  +body: string;
+  +icon: string;
+  +level: NotificationLevel;
+  +linkTo: string;
+  +creator?: string;
+  +subtype?: string;
 }
 
-export const PROPOSAL_NEW = ({ timestamp, data }: TemplateArguments): TemplateDescription => {
-  const avatar = ''
-  const member = `<strong>${data.creator}</strong>`
+const strong = (text) => `<strong>${text}</strong>`
 
-  const bodyMap = {
-    MINCOME: L('{member} proposed to change the group mincome. Vote now!', { member }), // @mmbotelho copy changed
-    ADD_MEMBER: L('{member} proposed to add members to the group. Vote now!', { member }),
-    REMOVE_MEMBER: L('{member} proposed to remove a member from the group. Vote now!', { member }),
-    VOTING_RULE: L('{member} proposed to change the group voting system. Vote now!', { member })
-  }
+export default ({
+  CONTRIBUTION_REMINDER (data: { monthstamp: string }) {
+    return {
+      // REVIEW @mmbotelho - Rename contributions to payments?
+      // TODO: Translate `month` when English is not the current language.
+      body: L('Do not forget to send your {month} contributions.', {
+        month: strong(monthName(data.monthstamp))
+      }),
+      icon: 'coins',
+      level: 'info',
+      linkTo: '/payments'
+    }
+  },
+  INCOME_DETAILS_OLD (data: {}) {
+    return {
+      body: L("You haven't updated your income details in more than 6 months. Would you like to review them now?"),
```

Comment:
Maybe make the `6 months` an argument? It seems like the kind of thing better kept as a constant.

Reply:
Yea that's a good idea. Every "reminder" notification type should likely be associated with some kind of configurable threshold constant. However, I'm not sure about the details yet.